flight/Modules/Battery/battery.c

```diff
@@ -128,18 +128,28 @@ static void batteryTask(void * parameters)
 				currentADCPin = -1;
 		}
 
+		bool adc_pin_invalid = false;
+		bool adc_offset_invalid = false;
+
 		// handle voltage
 		if (voltageADCPin >= 0) {
 			flightBatteryData.Voltage = (float)PIOS_ADC_GetChannelVolt(voltageADCPin);
 
 			// A negative result indicates an error (PIOS_ADC_GetChannelVolt returns -1 on error)
 			if(flightBatteryData.Voltage < 0.0f)
-				flightBatteryData.Voltage = 0.0f;
+				adc_pin_invalid = true;
```

Comment:
The batterySettings.SensorCalibrationFactor can also cause the voltage or current to be negative, so I wanted to check it after the scaling to avoid the situation where we pass on a negative value to downstream layers. The reason I check it before the scaling is to differentiate between an ADC error and the calibration offset causing it to go negative, you are right that the scaling doesn't need to go ahead when there is an ADC error so I will look at cleaning it up. Thanks. :)